src/interpret.ts

```diff
@@ -371,11 +371,15 @@ function createResultIndex(
     const id = response.artifactName;
     let sha: string | undefined = undefined;
     const shaPath = path.join(response.downloadPath, "sha.txt");
-    if (
-      fs.existsSync(shaPath) &&
-      fs.readFileSync(shaPath, "utf-8").length > 0
-    ) {
+    try {
       sha = fs.readFileSync(shaPath, "utf-8");
+    } catch (err: any) {
+      // check if this is a file not found error
+      if (err?.code === "ENOENT") {
+        console.log("File not found!");
+      } else {
+        throw err;
+      }
```

Comment:
@charisk what do you think here? With the exists check first CodeQL warned of a race condition, so we changed it to catch the exception if the file doesn't exist instead. On the other hand this block feels very clunky. Perhaps we should just keep it simple and swallow any exception?

Reply:
To me it looks like it doesn't matter what went wrong, but just that _something_ went wrong. So it'd make sense to try/catch any exception, maybe log it as a warning but continue normal operation. We'll then assume HEAD if the SHA isn't available in the extension code. 

If that turns out too problematic in practise we can change it.

Reply:
Thanks! Something like this? https://github.com/dsp-testing/qc-run2/pull/639/commits/fd875ca0fb62d00fa9b9fb86fa2d9f1e91d886b8